File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
bool isVowel(char c) {
    return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u'
        || c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U';
}

class Solution {
public:
    string reverseVowels(string s) {
        int i = 0, j = size(s) - 1;
        while (i < j) {
            while (i < j && !isVowel(s[i])) {
                i++;
            }
            while (i < j && !isVowel(s[j])) {
                j--;
            }
            swap(s[i], s[j]);
            i++, j--;
        }
        return s;
    }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public static String vowels = "aeiouAEIOU";
    
    public String reverseVowels(String s) {
        char[] charArray = s.toCharArray();
        int left = 0, right = charArray.length - 1;
        while (left < right) {
            while (left < right && !isVowel(charArray[left])) {
                left++;
            }
            while (left < right && !isVowel(charArray[right])) {
                right--;
            }
            char temp = charArray[left];
            charArray[left] = charArray[right];
            charArray[right] = temp;
            left++;
            right--;
        }
        return String.valueOf(charArray);
    }
    
    public boolean isVowel(char c) {
        return vowels.indexOf(c) != -1;
    }
}